docs/architecture-decisions/support-code-default.md

```diff
@@ -0,0 +1,333 @@
+   - Clear distinction from error cases (which return 4xx/5xx)
+
+   **flagd RPC**:
+   - Omit both the type-specific value field and variant field in response messages
+   - Use protobuf field presence to signal "no opinion from server"
+   - No changes needed to RPC method signatures
+
+5. **Provider Implementation**:
+   - Check for presence/absence of value field in responses
+   - When value is absent and reason is "DEFAULT", use code-defined default
+   - When value is present (even if null/false/empty), use that value
+   - Variant field will also be absent in delegation responses, resulting in undefined variant in resolution details
+   - Responses with value fields work as before, maintaining backward compatibility
+
+### Design Rationale
+
+**Using "DEFAULT" reason**: We intentionally reuse the existing "DEFAULT" reason code rather than introducing a new one (like "CODE_DEFAULT"). The distinction between a configuration default and code default is clear from the response structure:
+
+- Configuration default: Has both value and variant fields
+- Code default: Omits both value and variant fields
+
+**Field Omission Pattern**: Using field presence/absence is a well-established pattern in protocol design:
+
+- Unambiguous: Cannot confuse "null value" with "no opinion"
+- Language agnostic: Works across type systems
+- Protocol friendly: Natural in both JSON and Protobuf
+- Backward compatible: Existing responses always include values
+- Spec compliant: OpenFeature allows undefined variants
+
+The omission of both value and variant fields creates a clear, consistent signal that the server is fully delegating the decision to the client's code default.
+
+This approach maintains compatibility with the established OpenFeature terminology while providing clear semantics through response structure.
+
+### API changes
+
+**Flag Configuration**:
+
+```yaml
+flags:
+  my-feature:
+    state: ENABLED
+    defaultVariant: null  # Explicitly use code default
+    variants:
+      on: true
+      off: false
+    targeting:
+      if:
+        - "===":
+            - var: user-type
+            - "beta"
+        - on
+```
+
+**OFREP Response** when code default is indicated:
+
+#### Single flag evaluation response
+
+```json
+{
+  "key": "my-feature",
+  "reason": "DEFAULT",
+  "metadata": {}
+}
+```
+
+#### Bulk flag evaluation response
+
+```json
+{
+  "flags": [
+    {
+      "key": "my-feature",
+      "reason": "DEFAULT",
+      "metadata": {}
+    }
+  ]
+}
+```
+
+Note: Both `value` and `variant` fields are intentionally omitted to signal "use code default"
+
+**flagd RPC Response** (ResolveBooleanResponse):
+
+```protobuf
+{
+  "reason": "DEFAULT",
+  "metadata": {}
+}
+```
+
+Both the type-specific value field and variant field are omitted
+
+**Provider behavior**:
+
+1. Check response for presence of value field
+2. If value field is absent and reason is "DEFAULT", use code-defined default
```

Comment:
```suggestion
2. If value field is absent and error code is absent , use code-defined default
```

I _think_ this is the heuristic we want. Regardless of the DEFAULT reason. The provider can set DEFAULT itself in this case in the resolution, I think. :thinking: